src/main/java/persistence/sql/entity/PersistenceContextImpl.java

```diff
@@ -0,0 +1,57 @@
+package persistence.sql.entity;
+
+import jakarta.persistence.Id;
+
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+
+public class PersistenceContextImpl implements PersistenceContext {
+    private final Map<Class<?>, Map<Long, Object>> managedEntities = new HashMap<>();
+
+    @Override
+    public <T> T getEntity(Class<T> clazz, Long id) {
+        if (!containsEntity(clazz, id)) {
+            return null;
+        }
+        return clazz.cast(managedEntities.get(clazz).get(id));
+    }
+
+    @Override
+    public void addEntity(Object entity) {
+        Class<?> clazz = entity.getClass();
+        Long idValue = getIdValue(entity);
```

Comment:
영속성 컨텍스트에서 리플렉션을 통해 id를 찾아내는 대신 메서드 파라미터로 전달 받는 방법은 어떨까요? :)

```suggestion
    @Override
    public void addEntity(Object entity, Long id) {
        Class<?> clazz = entity.getClass();
```

Reply:
이부분도 테스트를 작성하려다가 id를 같이주는게 좋을것같다고 생각했었습니다. 👍